client/my-sites/email/email-management/home/email-plan/index.jsx

```diff
@@ -121,19 +123,72 @@ function EmailPlan( { domain, hideHeaderCake = false, selectedSite, source } ) {
 		);
 	};
 
-	const addEmailForwardMutationActive = useAddEmailForwardMutationIsLoading();
+	// Email checkout provides the /mailbox route with a new email param, e.g. /mailboxes/example.com?new-email=example@example.com
+	const queryParams = new URLSearchParams( window.location.search );
+	const newEmail = queryParams.get( 'new-email' );
+
+	const refetchTimeoutRef = useRef( null );
+	const [ refetchCount, setRefetchCount ] = useState( 0 );
 
-	const { data: emailAccounts = [], isLoading } = useGetEmailAccountsQuery(
-		selectedSite.ID,
-		domain.name,
-		{
-			refetchOnMount: ! addEmailForwardMutationActive,
-			retry: false,
+	useEffect( () => {
+		if ( ! newEmail ) {
+			return;
 		}
-	);
+		dispatch(
+			successNotice( translate( 'Your mailbox has been created.' ), {
+				duration: 8000,
+			} )
+		);
+	}, [ newEmail, dispatch, translate ] );
+
+	const addEmailForwardMutationActive = useAddEmailForwardMutationIsLoading();
+
+	const {
+		data: emailAccounts = [],
+		isLoading: isLoadingEmailAccounts,
+		refetch,
+	} = useGetEmailAccountsQuery( selectedSite.ID, domain.name, {
+		refetchOnMount: ! addEmailForwardMutationActive,
+		retry: false,
+	} );
 
 	const emailForwardsLimit = getEmailForwardLimit( emailAccounts );
 
+	const mailbox = newEmail ? newEmail.split( '@' )[ 0 ] : '';
+	const emailExists = emailAccounts.some( ( account ) =>
+		account.emails.some( ( email ) => {
+			return mailbox === email.mailbox;
+		} )
+	);
+	const isLoading = isLoadingEmailAccounts || ( newEmail && ! emailExists && refetchCount < 5 );
+
+	// Email provisioning takes a few seconds to complete, if there is a newEmail
+	// Refetch email acounts every 1.5 seconds up to 5 times until we can see it
+	useEffect( () => {
+		if ( ! newEmail || emailExists ) {
+			return;
+		}
+
+		if ( refetchCount >= 5 ) {
+			return;
+		}
+
+		if ( refetchTimeoutRef.current ) {
+			clearTimeout( refetchTimeoutRef.current );
+		}
+
+		refetchTimeoutRef.current = setTimeout( () => {
+			refetch();
+			setRefetchCount( ( prev ) => prev + 1 );
+		}, 1500 );
+
+		return () => {
+			if ( refetchTimeoutRef.current ) {
+				clearTimeout( refetchTimeoutRef.current );
+			}
+		};
+	}, [ newEmail, refetch, emailAccounts, emailExists, refetchCount ] );
```

Comment:
```suggestion
	}, [ newEmail, refetch, emailExists, refetchCount ] );
```
I'm guessing this is a leftover